flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/exceptions.scala

```diff
@@ -37,3 +37,16 @@ case class ValidationException(msg: String) extends RuntimeException(msg)
   * Exception for unwanted method calling on unresolved expression.
   */
 case class UnresolvedException(msg: String) extends RuntimeException(msg)
+
+/**
+  * Exceptions for all errors occurring during sql parse phase
+  */
+case class FlinkSqlParseException(msg: String) extends RuntimeException(msg)
```

Comment:
I think we should not include Flink in every class name. `SqlParserException` would be better to be consistent with `ExpressionParserException`.